### ISSUE TITLE
 A value of type 'SecurityContext/*1*/' can't be assigned to a variable of type 'SecurityContext/*2*/'

### ISSUE THREAD

**Issue by RicardoRB** [dint-dev/universal_io#7]
Hi!

I'm building my project for web and I have an error with SecurtyContext but no idea how to resolve it because it is complaining about types.

```
Target dart2js failed: Exception: lib/courier/bloc/courier_bloc.dart:78:32:
Error: A value of type 'SecurityContext/*1*/' can't be assigned to a variable of type 'SecurityContext/*2*/'.
 - 'SecurityContext/*1*/' is from 'package:universal_io/src/io_impl_js/security_context.dart'
 ('../../flutter/.pub-cache/hosted/pub.dartlang.org/universal_io-1.0.1/lib/src/io_impl_js/security_context.dart').
 - 'SecurityContext/*2*/' is from 'dart:io'.
      client.securityContext = securityContext;
                               ^
lib/location/bloc/delivery_location_bloc.dart:65:32:
Error: A value of type 'SecurityContext/*1*/' can't be assigned to a variable of type 'SecurityContext/*2*/'.
 - 'SecurityContext/*1*/' is from 'package:universal_io/src/io_impl_js/security_context.dart'
 ('../../flutter/.pub-cache/hosted/pub.dartlang.org/universal_io-1.0.1/lib/src/io_impl_js/security_context.dart').
 - 'SecurityContext/*2*/' is from 'dart:io'.
      client.securityContext = securityContext;
                               ^

```

**Comment by Naiki** [comment]
Just encountered the same with type File. The Problem by me was that it wanted to convert the universal io Filetype to a Widget which uses dart ui File type. 
I solved this problem with a typecast.
`import 'dart:io' as io;`
 `FileImage(_imageFiles[imageIndex] as io.File)`

Maybe that helps you or another person to find a faster solution to your problem.

**Comment by jlubeck** [comment]
Im' having the same issue but with `X509Certificate`

```
lib/api/api.dart:27:12: Error: A value of type 'bool Function(X509Certificate/*1*/, String, int)' can't be assigned to a variable of type 'bool Function(X509Certificate/*2*/, String, int)'.
 - 'X509Certificate/*1*/' is from 'package:universal_io/src/io_impl_js/secure_socket.dart' ('../../../.pub-cache/hosted/pub.dartlang.org/universal_io-1.0.2/lib/src/io_impl_js/secure_socket.dart').
 - 'X509Certificate/*2*/' is from 'dart:io'.
          ((X509Certificate cert, String host, int port) => Platform.isAndroid);
           ^
```

this is the code I'm using

```dart
  static http.Client get client {
    RetryPolicy retryPolicy = ApiRetryPolicy();
    var httpClient = HttpClientWithInterceptor.build(
        retryPolicy: retryPolicy, interceptors: []);
    if (env.useProxy == true) {
      String proxy =
          Platform.isAndroid ? '<YOUR_LOCAL_IP>:8888' : 'localhost:8888';

      httpClient.findProxy = (uri) {
        return "PROXY $proxy;";
      };

      // This is a workaround to allow Charles to receive
      // SSL payloads when your app is running on Android.
      httpClient.badCertificateCallback =
          ((X509Certificate cert, String host, int port) => Platform.isAndroid);

      return httpClient;
    }
    return httpClient;
  }

```

**Comment by terrier989** [comment]
Usually this can be solved by using `package:universal_io/io.dart` rather than `package:universal_io/prefer_universal/io.dart`. The latter should probably be deprecated.

**Comment by glenfordwilliams** [comment]
having a similar issue  when using import `'package:universal_io/io.dart'`

```
 Error: A value of type 'File/*1*/' can't be assigned to a variable of type 'File/*2*/'.
 - 'File/*1*/' is from 'dart:io'.                                       
 - 'File/*2*/' is from 'package:universal_io/src/io_impl_js/file.dart' ('../../../../.pub-cache/hosted/pub.dartlang.org/universal_io-1.0.2/lib/src/io_impl_js/file.dart').
    File croppedFile = await ImageCropper.cropImage(    
```

any solution?

**Comment by ayushg162** [comment]
> @glenfordwilliams I am facing same issue. Have you solved?

**Comment by terrier989** [comment]
Hi! Thanks for the issue reports. These are fixed in 2.0.2.
